src/bucket.ts

```diff
@@ -465,6 +467,39 @@ export class S3Bucket {
       deleteMarker: resp.headers.get("x-amz-delete-marker") === "true",
     };
   }
+
+  async empty(): Promise<DeletedKeys> {
+    const deleted: DeletedKeys = [];
+    let ls: ListObjectsResponse | undefined;
+    do {
+      ls = await this.listObjects({
+        maxKeys: 20,
+        continuationToken: ls?.nextContinuationToken,
+      });
+
+      const d = await this.deleteMany(ls?.contents ?? []);
+      deleted.push(...d);
+    } while (ls?.nextContinuationToken);
+    return deleted;
+  }
+
+  private async deleteMany(objects: S3Object[]): Promise<DeletedKeys> {
+    const p: Promise<DeleteObjectResponse>[] = [];
+    const deleted: DeletedKeys = [];
+    objects.forEach((o) => {
```

Comment:
Can you use pooledMap here (std/async/pool.ts)? Sending out too many requests at once will crash Deno or cause s3 to reject requests.

Reply:
yeah, been there in the past with Node (usually node crashes before s3 returns an error) that's why I used maxKeys 20 on the listObject, but I hadn't thought to use the async pool, will give it a look